### PR TITLE
feat: support for persisted queries

### PR DESCRIPTION
**Pull request by salmanm**
First draft for https://github.com/mcollina/fastify-gql/issues/110

- Added support to accept `persistedQueries` as the plugin options. It's a JSON object containing hash -> query mapping which is used by GET queries handler to resolve to full query text.

### CODE REVIEW

File: examples/persisted-queries/queries.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "1111": "{ add(x: 1, y: 1) }",
+  "2222": "query Add($x: Int!, $y: Int!) { add(x: $x, y: $y) }",
+  "3333": "{ add(x: 3, y: 3) }"
```
Comment (mcollina) [inline examples/persisted-queries/queries.json:4]: Can you use proper hashes?

File: lib/routes.js [hunk 2]
```diff
@@ -118,6 +119,9 @@ module.exports = async function (app, opts) {
           query: {
             type: 'string'
           },
+          isPersisted: {
```
Comment (mcollina) [inline lib/routes.js:122]:
```suggestion
          persisted: {
```

**Comment by salmanm** [comment]
Added option `onlyPersisted`. Also added some tests.

**Comment by frikille** [comment]
Shouldn't persisted queries be supported via `POST` as well? It would increase security by rejecting unknown queries.

**Comment by salmanm** [comment]
Good point. The initial plan wasn't to make the server reject unknown queries, but now it is. So it'd make sense to do so on post too. Updating.

File: README.md [hunk 3]
```diff
@@ -91,6 +91,39 @@ app.get('/', async function (req, reply) {
 app.listen(3000)
 ```
 
+### persistedQueries support
+
```
Comment (thomheymann) [inline README.md:95]: I would add a quick explanation about what persisted queries are and why they're useful

Comment (salmanm) [reply README.md:95]: Yep, that'd be useful. Done.

**Pull request merged**